File: howManyDalmations.js
function howManyDalmatians(num) {
    const dogs = ["Hardly any", "More than a handful!", "Woah that's a lot of dogs!", "101 DALMATIANS!!!"];
    return (num <= 10) ? dogs[0] : (num <= 50) ? dogs[1] : (num == 101) ? dogs[3] : dogs[2];
}

Test.assertEquals(howManyDalmatians(26), "More than a handful!");
Test.assertEquals(howManyDalmatians(8), "Hardly any");
Test.assertEquals(howManyDalmatians(14), "More than a handful!");
Test.assertEquals(howManyDalmatians(80), "Woah that's a lot of dogs!");
Test.assertEquals(howManyDalmatians(100), "Woah that's a lot of dogs!");
Test.assertEquals(howManyDalmatians(101), "101 DALMATIANS!!!");

/* FIX THE FOLLOWING CODE BLOCK
function howManyDalmations(numer :
  
  var dogs ["Hardly any", "More than a handful!", "Woah that's a lot of dogs!", "101 DALMATIONS!!!"];
  
  var respond = number <= 10 ? dogs[0] (number <= 50 ? dogs[1] : (number = 101  dogs[3] : dogs[2]
  
return respond

Test.describe("Basic tests",_=>{
Test.assertEquals(howManyDalmatians(26), "More than a handful!");
Test.assertEquals(howManyDalmatians(8), "Hardly any");
Test.assertEquals(howManyDalmatians(14), "More than a handful!");
Test.assertEquals(howManyDalmatians(80), "Woah that's a lot of dogs!");
Test.assertEquals(howManyDalmatians(100), "Woah that's a lot of dogs!");
Test.assertEquals(howManyDalmatians(101), "101 DALMATIANS!!!");
})
*/

/* SOLUTION:
There are several things wrong with the code,
stemming from everything improper closure to typos
and misspellings. I went through, simplified and fixed
the issues.
*/

File: howManyDalmations.py
def howManyDalmatians(num):
    dogs = ["Hardly any", "More than a handful!", "Woah that's a lot of dogs!", "101 DALMATIANS!!!"]
    if (num <= 10):
        return dogs[0] 
    elif (num <= 50):
        return dogs[1]
    elif (num == 101):
        return dogs[3]
    else:
        return dogs[2]

print(howManyDalmatians(26))
print(howManyDalmatians(8))
print(howManyDalmatians(14))
print(howManyDalmatians(80))
print(howManyDalmatians(100))
print(howManyDalmatians(50))
print(howManyDalmatians(10))
print(howManyDalmatians(101))

""" FIX ALL THE ERRORS IN THIS CODE
def how_many_dalmatians(n):
  dogs ["Hardly any", "More than a handful!", "Woah that's a lot of dogs!", "101 DALMATIONS!!!"];
  
  respond = if number <= 10 then dogs[0] else if (number <= 50 then dogs[1] else (number = 101  dogs[3] else dogs[2]
  
return respond
"""